crates/bin/prove_block/src/lib.rs

```diff
@@ -131,21 +131,21 @@ pub async fn prove_block(
     };
 
     // We only need to get the older block number and hash. No need to fetch all the txs
-    let older_block = match rpc_client
-        .starknet_rpc()
-        .get_block_with_tx_hashes(BlockId::Number(block_number - STORED_BLOCK_HASH_BUFFER))
-        .await?
-    {
-        MaybePendingBlockWithTxHashes::Block(block_with_txs_hashes) => block_with_txs_hashes,
-        MaybePendingBlockWithTxHashes::PendingBlock(_) => {
-            panic!("Block is still pending!");
-        }
-    };
-
-    let block_context = build_block_context(chain_id.clone(), &block_with_txs, starknet_version);
-
+    // This is a workaorund to catch the case where the block number is less than the buffer and still preserve the check
+    // The OS will also handle the case where the block number is less than the buffer.
+    let older_block_number =
+        if block_number <= STORED_BLOCK_HASH_BUFFER { 1 } else { block_number - STORED_BLOCK_HASH_BUFFER };
```

Comment:
This is what the OS does for a similar check:

```
    tempvar old_block_number = block_context.block_info_for_execute.block_number -
        STORED_BLOCK_HASH_BUFFER;
    let is_old_block_number_non_negative = is_nn(old_block_number);
    if (is_old_block_number_non_negative == FALSE) {
        // Not enough blocks in the system - nothing to write.
        return ();
    }
```

What we do in `prove_block` with this is feed it to `ExecutionHelper`, which is used later by the OS just after the code snippet above to set the block number/hash. Since it's protected by the check above, I think it should be ok.

However, it may be cleaner to make `old_block_number` and `old_block_hash` Optional, and it would also be good to avoid doing an RPC query in this case.